### PR TITLE
Use numpy isclose comparison in hermiticity check

### PR DESCRIPTION
**Pull request by jwnys**
Hermiticity check was too strong, but uses numpy.isclose now with some tolerance.

### CODE REVIEW

File: netket/experimental/operator/_fermions_2nd.py [hunk 1]
```diff
@@ -669,10 +669,34 @@ def _check_hermitian(
...
+    """Compare two dicts and return True if their keys and values are all the same (up to some tolerance)"""
+    d1_keys = set(d1.keys())
+    d2_keys = set(d2.keys())
+    shared_keys = d1_keys.intersection(d2_keys)
+    added = d1_keys - d2_keys
+    if len(added) != 0:
+        return False
+    removed = d2_keys - d1_keys
+    if len(removed) != 0:
+        return False
```
Comment (PhilipVinc) [inline netket/experimental/operator/_fermions_2nd.py:689]: Can't you just do the more readable
```
if not d1_keys == d2_keys:
    return False
```
?
Or this does not work for some reason?

**Comment by jwnys** [comment]
@PhilipVinc , it was actually an attempt to make it as readable as possible, but here's just the short version.

**Comment by jwnys** [comment]
@PhilipVinc , this is short and clear.

File: netket/experimental/operator/_fermions_2nd.py [hunk 2]
```diff
@@ -669,10 +669,23 @@ def _check_hermitian(
...
 
 
+def dict_compare(d1, d2):
+    """Compare two dicts and return True if their keys and values are all the same (up to some tolerance)"""
+    d1_keys = set(d1.keys())
+    d2_keys = set(d2.keys())
+    if d1_keys != d2_keys:
+        return False
+    shared_keys = d1_keys.intersection(d2_keys)
+    return all(np.isclose(d1[o], d2[o]) for o in shared_keys)
```
Comment (PhilipVinc) [inline netket/experimental/operator/_fermions_2nd.py:686]:
```suggestion
    # We checked that d1 and d2 have the same keys. Now check the values.
    return all(np.isclose(d1[o], d2[o]) for o in d1_keys)
```

**Review by PhilipVinc — APPROVED** [review]

**Comment by PhilipVinc** [comment]
Last nitpick: can you add a line to the CHANGELOG mentioning this change?

**Comment by jwnys** [comment]
I added:
* Use np.isclose to allow for some tolerance in checking the hermicity of a `nkx.operator.FermionOperator2nd` [#1233](https://github.com/netket/netket/pull/1233)
To the changelog

**Comment by PhilipVinc** [comment]
great thanks.
merging once this pass tests

**Pull request merged**